Simplify traffic and seo

JevgenijVisockij opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplify traffic and seo options form
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes 
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  |  Simplify Configure -> Shop Paramateres -> Traffic & SEO. Everything must look the same with two exceptions. Blue help boxes were replaced with help text under input. Also some fields now are marked as required, because they always were required.


:notebook: **BC Break**
Backwards compatibility break introduced due to extension of TranslationAwareType by SEOOptionsType, SetUpUrlType, ShopUrlType, UrlSchemaType. This means if any module extends any of those types they will get an exception. 
<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/20243)
<!-- Reviewable:end -->

JevgenijVisockij committed:
CS Fixer fixes

JevgenijVisockij commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php:
```diff
@@ -52,44 +53,92 @@ class SetUpUrlType extends AbstractType
+        if (!$this->tools->isModRewriteActive()) {
+            $friendlyUrlHelp .=
+                '</br>' . $this->trans(
+                'URL rewriting (mod_rewrite) is not active on your server, or it is not possible to check your server configuration. If you want to use Friendly URLs, you must activate this mod.',
```
</br> works as intended but if there is a better or cleaner way to add new line I would love to hear a suggestion.

Julievrz replied:
@JevgenijVisockij is it normal that there is no "e" at mod? :)

JevgenijVisockij replied:
you mean "mod" in 

> you must activate this mod.

Right? In that case, yeah it seems so. String was exactly the same before and the apache module is called "mod_rewrite"

Julievrz replied:
Ok thank you I just wanted to be sure :)

JevgenijVisockij commented on src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml:
```diff
@@ -603,8 +608,17 @@ services:
         class: 'PrestaShopBundle\Form\Admin\Configure\ShopParameters\TrafficSeo\Meta\UrlSchemaType'
+        parent: 'form.type.translatable.aware'
+        public: true
         arguments:
```
Couldn't figure out why this argument was here in the first place, I guess it was somesort of leftover.

JevgenijVisockij commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/UrlSchemaType.php:
```diff
@@ -62,4 +126,34 @@ public function configureOptions(OptionsResolver $resolver)
             'label' => false,
         ]);
     }
+
```
Logic taken from
PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/TrafficSeo/Meta/Blocks/keyword.html.twig

matks commented:
PHPStan not happy 😅

JevgenijVisockij committed:
Fixed PHPStan issue

atomiix commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php:
```diff
@@ -52,44 +54,92 @@ class SetUpUrlType extends AbstractType
+
+        if (!$this->tools->isModRewriteActive()) {
+            $friendlyUrlHelp .=
+                '</br>' . $this->trans(
```
I don't have a cleaner way, but here is a suggestion to be W3C compliant 😄
```suggestion
                '<br>' . $this->trans(
```
Or
```suggestion
                '<br/>' . $this->trans(
```

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php

Changed the slash of br to the other side.

Co-authored-by: atomiix <tbaccelli@gmail.com>

Julievrz commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php:
```diff
@@ -52,44 +54,92 @@ class SetUpUrlType extends AbstractType
+                'help' => $friendlyUrlHelp,
+            ])
+            ->add('accented_url', SwitchType::class, [
+                'label' => $this->trans('Accented URL', 'Admin.Global'),
```
"Accented URL" should be located in Admin.Shopparameters.Feature

Julievrz commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php:
```diff
@@ -52,44 +54,92 @@ class SetUpUrlType extends AbstractType
                 [
                     'choices' => $this->canonicalUrlChoices,
                     'translation_domain' => false,
+                    'label' => $this->trans('Redirect to the canonical URL', 'Admin.Global'),
```
This wording should be located in Admin.Shopparameters.Feature

Julievrz commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php:
```diff
@@ -52,44 +54,92 @@ class SetUpUrlType extends AbstractType
-                ->add('disable_apache_multiview', SwitchType::class)
-                ->add('disable_apache_mod_security', SwitchType::class);
+                ->add('disable_apache_multiview', SwitchType::class, [
+                    'label' => $this->trans('Disable Apache\'s MultiViews option', 'Admin.Global'),
```
This wording should be located in Admin.Shopparameters.Feature

Julievrz commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php:
```diff
@@ -52,44 +54,92 @@ class SetUpUrlType extends AbstractType
+                    ),
+                ])
+                ->add('disable_apache_mod_security', SwitchType::class, [
+                    'label' => $this->trans('Disable Apache\'s mod_security module', 'Admin.Global'),
```
This wording should be located in Admin.Shopparameters.Feature

LouiseBonnard commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/UrlSchemaType.php:
```diff
@@ -26,31 +26,95 @@
+            ])
+            ->add('layered_rule', TextType::class, [
+                'label' => $this->trans(
+                    'Route to category which has the "selected_filter" attribute for the "Layered Navigation" (blocklayered) module',
```
⚠️ the _Layered Navigation_ module does not exist in the 1.7.x versions anymore. We should make sure it works with the _Faceted Search_ module and adapt the wording accordingly. ;-)

matks replied:
According to @JevgenijVisockij :
"As far as I checked the "layered_rule" isin't really used in faceted search. For that matter it's only used in Link::getCategoryLink function and I couldn't find where this function would be used to retrieve category with selected filters."

So my understanding of the situation is:
- this setting in PS 1.7 is intended for blocklayered module
- ps_facetedsearch is the successor of blocklayered module, but it does not use this setting

So maybe we could simply ... remove this 😄 ?

However we need to confirm "ps_facetedsearch does not use this setting"

@PierreRambaud can you confirm ? does it ring a bell ?

matks replied:
@LouiseBonnard I checked the code of ps_facetedsearch.
I confirm that `selected_filter` is a parameter that was usable with blocklayered but is not implement into ps_facetedsearch. We can delete this input, it's useless with ps_facetedsearch.

LouiseBonnard replied:
Perfect, thank you very much @matks & @JevgenijVisockij. Let's delete this input then! 😄

LouiseBonnard commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php:
```diff
@@ -52,44 +54,92 @@ class SetUpUrlType extends AbstractType
+            ->add('accented_url', SwitchType::class, [
+                'label' => $this->trans('Accented URL', 'Admin.Global'),
+                'help' => $this->trans(
+                    'Enable this option if you want to allow accented characters in your friendly URLs. You should only activate this option if you are using non-latin characters ; for all the latin charsets, your SEO will be better without this option.',
```
@JevgenijVisockij, did you decide to reunite the two translation strings into a single one? I'm okay with this option if @Julievrz is, it could be the occasion to fix a few typos, cf. the suggestion below.

LouiseBonnard replied:
```suggestion
                    'Enable this option if you want to allow accented characters in your friendly URLs. You should only activate this option if you are using non-Latin characters; for all the Latin charsets, your SEO will be better without this option.',
```

JevgenijVisockij replied:
@LouiseBonnard 

This one confused me for a second. The reason I changed this is because in this case I copied the string from BO instead of code. Well turns out by default in english

> Enable this option if you want to allow accented characters in your friendly URLs.

 is translated to 

> Enable this option if you want to allow accented characters in your friendly URLs. You should only activate this option if you are using non-latin characters ; for all the latin charsets, your SEO will be better without this option.

So I can fix this to be like original, or I can change to reflect the english translation with changes you mentioned.

Julievrz replied:
I think it is better to keep your suggestion (1 string that combines both sentences) and apply @LouiseBonnard's suggestions. :) Thank you @JevgenijVisockij !

JevgenijVisockij replied:
Alright, will do that!

LouiseBonnard commented:
> This pull request seems to contain new translation strings. I have summarized them below to ease up review:
> 
> * `Admin.Shopparameters.Feature`
>   
>   * [x]  [`Keywords: %s`](https://github.com/PrestaShop/PrestaShop/pull/21472/files#diff-75732d2a6decda5a30acb8d99b717adfR153)
> * `Admin.Shopparameters.Help`

@atomiix, this wording is considered as new by PrestonBot but I can see on Crowdin that it was part of the 1.7.2 translation catalog. Is it normal?

<img width="598" alt="Capture d’écran 2020-10-16 à 17 05 25" src="https://user-images.githubusercontent.com/32565890/96275460-0c5e7d80-0fd2-11eb-899d-2b36b86ce0e7.png">

LouiseBonnard commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/UrlSchemaType.php:
```diff
@@ -62,4 +126,34 @@ public function configureOptions(OptionsResolver $resolver)
+
+        return sprintf(
+            $this->trans(
+                'Keywords: %s',
```
@JevgenijVisockij, it is likely that it should be replaced by _Keywords: %keywords%_ (still localized in _Admin.Shopparameters.Feature_), this string does not exist since the 1.7.3 catalog, cf. https://github.com/PrestaShop/PrestaShop/pull/21472#issuecomment-710103625

JevgenijVisockij committed:
Fixed the domain for some translations

JevgenijVisockij committed:
Fixed the variable in translation

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php

Co-authored-by: LouiseBonnard <32565890+LouiseBonnard@users.noreply.github.com>

JevgenijVisockij committed:
CS Fixer fixes

atomiix commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/ShopUrlType.php:
```diff
@@ -59,8 +60,13 @@ class ShopUrlType extends AbstractType
-    {
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales, $isHostMode,
```
```suggestion
        array $locales,
        $isHostMode,
```

You'll need to update the phpdoc too ;)

JevgenijVisockij committed:
Updated PHPDoc and fixed issue in argument being in the same row

JevgenijVisockij commented:
WIP because need to figure out this https://github.com/PrestaShop/PrestaShop/pull/21472#discussion_r509356921

JevgenijVisockij committed:
Removed setting that is no longer needed

matks commented:
> WIP because need to figure out this [#21472 (comment)](https://github.com/PrestaShop/PrestaShop/pull/21472#discussion_r509356921)

We have decided to delete it ✅

JevgenijVisockij committed:
CS fixer fix

JevgenijVisockij committed:
Added return time to renderForm

JevgenijVisockij committed:
Added comment to renderForm function

JevgenijVisockij commented:
> > WIP because need to figure out this [#21472 (comment)](https://github.com/PrestaShop/PrestaShop/pull/21472#discussion_r509356921)
> 
> We have decided to delete it white_check_mark

Yeah, I need to do some more changes but now PR is ready for review.

JevgenijVisockij committed:
CS fixer fix

atomiix commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/ShopUrlType.php:
```diff
@@ -55,12 +56,20 @@ class ShopUrlType extends AbstractType
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        $isHostMode,
+        $isShopFeatureActive,
+        $doesMainShopUrlExist
```
```suggestion
        bool $isHostMode,
        bool $isShopFeatureActive,
        bool $doesMainShopUrlExist
```

atomiix commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php:
```diff
@@ -52,44 +54,92 @@ class SetUpUrlType extends AbstractType
         array $canonicalUrlChoices,
         $isHtaccessFileWritable,
-        $isHostMode
+        $isHostMode,
+        $tools
```
```suggestion
        bool $isHtaccessFileWritable,
        bool $isHostMode,
        bool $isModRewriteActive
```

atomiix commented on src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php:
```diff
@@ -52,44 +54,92 @@ class SetUpUrlType extends AbstractType
      * @param array $canonicalUrlChoices
      * @param bool $isHtaccessFileWritable
      * @param bool $isHostMode
+     * @param Tools $tools
```
```suggestion
     * @param bool $isModRewriteActive
```

Progi1984 commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php:
```diff
@@ -54,81 +55,23 @@ class MetaController extends FrameworkBundleAdminController
         $setUpUrlsForm = $this->getSetUpUrlsFormHandler()->getForm();
         $shopUrlsForm = $this->getShopUrlsFormHandler()->getForm();
         $seoOptionsForm = $this->getSeoOptionsFormHandler()->getForm();
+        $isRewriteSettingEnabled = $this->get('prestashop.adapter.legacy.configuration')->get('PS_REWRITING_SETTINGS');
```
```suggestion
        $isRewriteSettingEnabled = $this->get('prestashop.adapter.legacy.configuration')->getBoolean('PS_REWRITING_SETTINGS');
```

Progi1984 commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php:
```diff
@@ -286,103 +229,108 @@ public function deleteBulkAction(Request $request)
         );
+        $shopUrlsForm = $this->getShopUrlsFormHandler()->getForm();
+        $seoOptionsForm = $this->getSeoOptionsFormHandler()->getForm();
+        $isRewriteSettingEnabled = $this->get('prestashop.adapter.legacy.configuration')->get('PS_REWRITING_SETTINGS');
```
```suggestion
        $isRewriteSettingEnabled = $this->get('prestashop.adapter.legacy.configuration')->getBoolean('PS_REWRITING_SETTINGS');
```

Progi1984 commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php:
```diff
@@ -286,103 +229,108 @@ public function deleteBulkAction(Request $request)
             'ShopUrls'
         );
+        $seoOptionsForm = $this->getSeoOptionsFormHandler()->getForm();
+        $isRewriteSettingEnabled = $this->get('prestashop.adapter.legacy.configuration')->get('PS_REWRITING_SETTINGS');
```
```suggestion
        $isRewriteSettingEnabled = $this->get('prestashop.adapter.legacy.configuration')->getBoolean('PS_REWRITING_SETTINGS');
```

Progi1984 commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php:
```diff
@@ -286,103 +229,108 @@ public function deleteBulkAction(Request $request)
-        if ($form->isSubmitted()) {
-            $data = $form->getData();
-            $saveErrors = $formHandler->save($data);
+        $isRewriteSettingEnabled = $this->get('prestashop.adapter.legacy.configuration')->get('PS_REWRITING_SETTINGS');
```
```suggestion
        $isRewriteSettingEnabled = $this->get('prestashop.adapter.legacy.configuration')->getBoolean('PS_REWRITING_SETTINGS');
```

Progi1984 reviewed (commented):
Some feedback from @atomiix and me

JevgenijVisockij commented:
> Some feedback from @atomiix and me

Thanks for the feedback! Impelmented it.

JevgenijVisockij committed:
Changed traffic and seo so it doesn't redirect after saving the form

# Conflicts:
#	src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php

JevgenijVisockij committed:
Implemented changes after code review

# Conflicts:
#	src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/Meta/SetUpUrlType.php
#	src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

JevgenijVisockij committed:
Removed $tools as it's no longer used

atomiix commented on src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/TrafficSeo/Meta/Blocks/set_up_urls_configuration.html.twig:
```diff
@@ -33,78 +33,7 @@
-        {% block meta_set_up_urls_form_rest %}
-          {{ form_rest(setUpUrlsForm) }}
-        {% endblock %}
+          {{ form_widget(setUpUrlsForm) }}
```
```suggestion
        {{ form_widget(setUpUrlsForm) }}
```

atomiix reviewed (commented):
Just a little cosmetic detail to fix and we should be good :)

SD1982 commented:
Hi @JevgenijVisockij it seems there is some conflicting files can you fix it please ?

JevgenijVisockij committed:
Update src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/TrafficSeo/Meta/Blocks/set_up_urls_configuration.html.twig

Removed unneeded space

Co-authored-by: atomiix <tbaccelli@gmail.com>

JevgenijVisockij committed:
Simplified traffic and seo options

# Conflicts:
#	src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/TrafficSeo/Meta/Blocks/shop_urls_configuration.html.twig

# Conflicts:
#	src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/TrafficSeo/Meta/Blocks/shop_urls_configuration.html.twig
#	src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/TrafficSeo/Meta/Blocks/url_schema_configuration.html.twig

JevgenijVisockij committed:
Added isModRewritableParameter which was missing

JevgenijVisockij commented:
> Hi @JevgenijVisockij it seems there is some conflicting files can you fix it please ?

Done

SD1982 commented:
Hi @JevgenijVisockij everything looks good to me !!
Thanks !!

Progi1984 commented:
Thanks @JevgenijVisockij & @SD1982
